includes/class-wc-coupon.php

```diff
@@ -154,131 +322,503 @@ private function get_coupon_id_from_code( $code ) {
+	 */
+	public function set_excluded_product_categories( $excluded_product_categories ) {
+		$this->_meta_data['exclude_product_categories'] = $excluded_product_categories;
+	}
+
+	/**
+	 * Set if this coupon should excluded sale items or not.
+	 * @since  2.6.0
+	 * @param  bool $exclude_sale_items
+	 */
+	public function set_exclude_sale_items( $exclude_sale_items ) {
+		$this->_meta_data['exclude_sale_items'] = (bool) $exclude_sale_items;
+	}
+
+	/**
+	 * Set the minimum spend amount.
+	 * @since  2.6.0
+	 * @param  float $amount
+	 */
+	public function set_minimum_amount( $amount ) {
+		$this->_meta_data['minimum_amount'] = wc_format_decimal( $amount );
+	}
+
+	/**
+	 * Set the maximum spend amount.
+	 * @since  2.6.0
+	 * @param  float $amount
+	 */
+	public function set_maximum_amount( $amount ) {
+		$this->_meta_data['maximum_amount'] = wc_format_decimal( $amount );
+	}
+
+	/**
+	 * Set email restrictions.
+	 * @since  2.6.0
+	 * @param  array $emails
+	 */
+	public function set_email_restrictions( $emails ) {
+		$this->_meta_data['customer_email'] = array_map( 'sanitize_email', $emails );
+	}
+
+	/**
+	 * Set which users have used this coupon.
+	 * @since 2.6.0
+	 * @param array $used_by
+	 */
+	public function set_used_by( $used_by ) {
+		$this->_meta_data['used_by'] = array_filter( $used_by );
+	}
+
+	/*
+	|--------------------------------------------------------------------------
+	| CRUD methods
+	|--------------------------------------------------------------------------
+	|
+	| Methods which create, read, update and delete coupons from the database.
+	|
+	| A save method is included for convenience (chooses update or create based
+	| on if the order exists yet).
+	|
+	*/
+
+	/**
+	 * Reads an coupon from the database and sets its data to the class.
+	 * @since 2.6.0
+	 * @param  int $id
+	 */
+	public function read( $id ) {
+		if ( 0 === $id ) {
+			$this->_data['id'] = 0;
+			return;
+		}
+
+		$post_object = get_post( $id );
+
+		// Only continue reading if this coupon exists...
+		if ( empty( $post_object ) || empty( $post_object->ID ) ) {
+			$this->_data['id'] = 0;
+			return;
+		}
+
+		$coupon_id   = $this->_data['id'] = absint( $post_object->ID );
+
+		// Map standard coupon data
+		$this->set_code( $post_object->post_title );
+		$this->set_description( $post_object->post_excerpt );
+		$this->set_discount_type( get_post_meta( $coupon_id, 'discount_type', true ) );
+		$this->set_amount( get_post_meta( $coupon_id, 'coupon_amount', true ) );
+		$this->set_expiry_date( get_post_meta( $coupon_id, 'expiry_date', true ) );
+		$this->set_usage_count( get_post_meta( $coupon_id, 'usage_count', true ) );
+
+		// Map meta data
+		$individual_use = ( 'yes' === get_post_meta( $coupon_id, 'individual_use', true ) ? true : false );
+		$this->set_individual_use( $individual_use );
+		$product_ids = ( ! empty ( get_post_meta( $coupon_id, 'product_ids', true ) ) ? explode( ',', get_post_meta( $coupon_id, 'product_ids', true ) ) : array() );
```

Comment:
Looks like only if you pass a negative `$limit` to explode: http://php.net/manual/en/function.explode.php -- but it does simplify the code. Otherwise, by default, you get an array with one index with an empty value.